flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ResolvedSchema.java

```diff
@@ -163,9 +163,16 @@ public Optional<UniqueConstraint> getPrimaryKey() {
         return Optional.ofNullable(primaryKey);
     }
 
-    /** Returns the primary key indexes, if any, otherwise returns an empty array. */
+    /**
+     * Returns the primary key indexes in the {@link #toPhysicalRowDataType()}, if any, otherwise
+     * returns an empty array.
+     */
     public int[] getPrimaryKeyIndexes() {
-        final List<String> columns = getColumnNames();
+        final List<String> columns =
+                getColumns().stream()
+                        .filter(Column::isPhysical)
```

Comment:
I am curious
- when can we have more than one primary key? I assume this is an unenforced constraint
- If so should we have a test for multiples?
- if we can have more than one primary key ,should put the keys in index order? Or will they already be in order due to how they come back from getColumns().

Reply:
@davidradl Have you checked the interfaces? https://github.com/apache/flink/blob/960a253e8bd8c80d8fce1bd7eb55d6aab16936ca/flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ResolvedSchema.java#L162

One can not have multiple primary keys. You can also check SQL standard which says you can only have one PRIMARY KEY. That's where `PRIMARY` come from.

Reply:
Hi @dawidwys thanks for you quick response, that is why I was curious - we return an array when there can only be one primary key. I assume there can be never be 2 elements in this array.